R/caretStack.R

```diff
@@ -100,12 +123,90 @@ caretStack <- function(
     models = all.models,
     ens_model = model,
     error = model$results,
-    excluded_class_id = excluded_class_id
+    excluded_class_id = excluded_class_id,
+    original_features_included = which.var
   )
   class(out) <- "caretStack"
   out
 }
 
+#' @title Check the parameters use_original_features, new_X, and which.var are consistent
+#' @description Check the parameters use_original_features, new_X, and which.var are consistent
+#' when training a caretStack.
+#' @param use_original_features a logical indicating whether to include the original features in the stack.
+#' @param new_X a data.frame of new data to train the stack on.
+#' @param which.var a vector indicating which original features to include in the stack.
+#' @keywords internal
+check_use_original_features <- function(use_original_features, new_X, which.var) {
```

Comment:
I think we we combine the 2 parameters into one parameter, and require it to be always a character vector we can remove the 2 validation functions and just 2 1-2 lines of validation directly in `caretStack`.